cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -135,6 +140,7 @@ protected void doShutdown() {
         }
       }), getRetryStrategy());
     }
+    metricScheduledService.stopAndWait();
```

Comment:
I think you should stop the service first.

Reply:
Fixed